kafka-connect-ftp/src/main/scala/com/datamountaineer/streamreactor/connect/ftp/source/FtpMonitor.scala

```diff
@@ -53,77 +55,155 @@ class FtpMonitor(settings:FtpMonitorSettings, fileConverter: FileConverter) exte
     case FtpProtocol.SFTP => new FTPSClient()
   }
 
-  def requiresFetch(file: AbsoluteFtpFile, metadata: Option[FileMetaData]): Boolean = metadata match {
-    case None => logger.debug(s"${file.name} hasn't been seen before"); true
-    case Some(known) if known.attribs.size != file.ftpFile.getSize => {
-      logger.debug(s"${file.name} size changed ${known.attribs.size} => ${file.size}")
-      true
-    }
-    case Some(known) if known.attribs.timestamp != file.timestamp => {
-      logger.debug(s"${file.name} is newer ${known.attribs.timestamp} => ${file.timestamp}")
-      true
+  def requiresFetch(mode: MonitorMode, file: AbsoluteFtpFile, metadata: Option[FileMetaData]): Boolean = {
+    logger.info(s"${mode}")
+    if( mode != MonitorMode.UpdateSlice ) {
+      metadata match {
+        case None => logger.debug(s"${file.name} hasn't been seen before"); true
+        case Some(known) if known.attribs.size != file.ftpFile.getSize => {
+          logger.debug(s"${file.name} size changed ${known.attribs.size} => ${file.size}")
+          true
+        }
+        case Some(known) if known.attribs.timestamp != file.timestamp => {
+          logger.debug(s"${file.name} is newer ${known.attribs.timestamp} => ${file.timestamp}")
+          true
+        }
+        case _ => false
+      }
+    } else {
+      metadata match {
+        case None => logger.info(s"${file.name} hasn't been seen before"); true
+        case Some(known) if known.attribs.size != file.ftpFile.getSize => {
+          logger.info(s"${file.name} size changed ${known.attribs.size} => ${file.size}")
+          true
+        }
+        case Some(known) if known.attribs.timestamp != file.timestamp => {
+          logger.info(s"${file.name} is newer ${known.attribs.timestamp} => ${file.timestamp}")
+          true
+        }
+        case Some(known) if known.offset != file.ftpFile.getSize => {
+          logger.info(s"${file.name} was not completely read & committed ${known.offset} => ${file.ftpFile.getSize}")
+          true
+        }
+        case _ =>
+          logger.info(s" no fetch required")
+          false
+      }
```

Comment:
I think the else & the match case could be combined. If I'm not wrong, most of the cases are identical ? 

Something like :

```scala
(mode, metadata) match {
    case (MonitorMode.UpdateSlice, Some(known) => ...
}
```

Reply:
The idea is to remove (maybe in next commit), the legacy code. So I don't want to mix my code with the legacy code.

So this :
```
if( !isSliceMode ) {
      //ENA : legacy code
      metadata match {
        case None => logger.debug(s"${file.name} hasn't been seen before"); true
        case Some(known) if known.attribs.size != file.ftpFile.getSize => {
          logger.debug(s"${file.name} size changed ${known.attribs.size} => ${file.size}")
          true
        }
        case Some(known) if known.attribs.timestamp != file.timestamp => {
          logger.debug(s"${file.name} is newer ${known.attribs.timestamp} => ${file.timestamp}")
          true
        }
        case _ => false
      }
    } else {
      metadata match {
        case None =>
          logger.info(s"${file.name} hasn't been seen before")
          true
        case Some(previous) if previous.attribs.size != file.ftpFile.getSize => {
          logger.info(s"${file.name} size changed ${previous.attribs.size} => ${file.size}")
          true
        }
        case Some(previous) if previous.attribs.timestamp != file.timestamp => {
          logger.info(s"${file.name} is newer ${previous.attribs.timestamp} => ${file.timestamp}")
          true
        }
        case Some(previous) if previous.offset != file.ftpFile.getSize => {
          logger.info(s"${file.name} was not completely read & committed ${previous.offset} => ${file.ftpFile.getSize}")
          true
        }
        case _ =>
          logger.info(s" no fetch required")
          false
      }
    }
```
would become this : 

```

      metadata match {
        case None =>
          logger.info(s"${file.name} hasn't been seen before")
          true
        case Some(previous) if previous.attribs.size != file.ftpFile.getSize => {
          logger.info(s"${file.name} size changed ${previous.attribs.size} => ${file.size}")
          true
        }
        case Some(previous) if previous.attribs.timestamp != file.timestamp => {
          logger.info(s"${file.name} is newer ${previous.attribs.timestamp} => ${file.timestamp}")
          true
        }
        case Some(previous) if previous.offset != file.ftpFile.getSize => {
          logger.info(s"${file.name} was not completely read & committed ${previous.offset} => ${file.ftpFile.getSize}")
          true
        }
        case _ =>
          logger.info(s" no fetch required")
          false
      }
```